src/charm.py

```diff
@@ -528,19 +521,19 @@ def _configure(self):  # noqa: C901
             v12_migration_date=v12_migration_date,
         ).build()
 
-        restart = self._update_config(config) or restart
+        restart = self._update_config(config)
```

Comment:
what does this restart variable represent? 
```suggestion
        has_config_changed = self._update_config(config)
```

Reply:
Yes your assumption is correct and I think the renaming of the variable is more clear now. Will include in the PR.